### PR TITLE
Upgrade to Go 1.22.1

### PR DESCRIPTION
**Pull request by maxlambrecht**
This PR upgrades the Go version to `1.22.1`, affecting the Dockerfile, Makefile, and GitHub Actions workflows (`nightly_build.yaml`, `pr_build.yaml`, `release_build.yaml`). It introduces the `.go-version` file to specify the Go version.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/spiffe/spire-controller-manager
 
-go 1.21
+go 1.22
```
Comment (azdagron) [inline go.mod:3]: We should probably specify the full version here (allowed since go1.21).

**Review by azdagron — COMMENTED** [review]
This might need rebasing against the latest in main. Marcos already made some of these changes.

**Comment by maxlambrecht** [comment]
I noticed that when building with the new Go version using `make build`, some CRD YAML files undergo modifications. It appears these changes were inadvertently included in my commit. Can you confirm if this is expected behavior?

File: go.mod (hunk 1 above)
Comment (maxlambrecht) [reply go.mod]: Done.

File: config/crd/bases/spire.spiffe.io_clusterspiffeids.yaml [hunk 2]
```diff
@@ -229,9 +242,3 @@ spec:
     storage: true
     subresources:
       status: {}
-status:
-  acceptedNames:
-    kind: ""
-    plural: ""
-  conditions: []
-  storedVersions: []
```
Comment (MarcosDY) [inline config/crd/bases/spire.spiffe.io_clusterspiffeids.yaml:237]: can you revert status removal? there are some cases where it is required,
but controller runtime is no longer generating this... (the same comment for all cases like this)

File: config/crd/bases/spire.spiffe.io_clusterfederatedtrustdomains.yaml [hunk 3]
```diff
@@ -3,7 +3,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.13.0
+    controller-gen.kubebuilder.io/version: v0.14.0
```
Comment (MarcosDY) [inline config/crd/bases/spire.spiffe.io_clusterfederatedtrustdomains.yaml:6]: can you update demo with this new generated yamls?  (same comment for all yaml)

File: go.mod [hunk 4]
```diff
@@ -1,6 +1,6 @@
 module github.com/spiffe/spire-controller-manager
 
-go 1.22
+go 1.22.1
```
Comment (MarcosDY) [inline go.mod:3]: why using 1.22.1? I think we must use 1.22

Comment (maxlambrecht) [reply go.mod:3]: https://github.com/spiffe/spire-controller-manager/pull/330#discussion_r1548528175

**Review by MarcosDY — APPROVED** [review]

**Pull request merged**